src/renderers/webgl/WebGLProgram.js

```diff
@@ -282,8 +282,21 @@ THREE.WebGLProgram = ( function () {
 
 		if ( material instanceof THREE.RawShaderMaterial ) {
 
-			prefixVertex = '';
-			prefixFragment = '';
+			prefixVertex = [
+
+				'#define SHADER_NAME ' + material.__webglShader.name,
```

Comment:
@mrdoob Well, I remembered the conversation... Took awhile to find it again, though... : - )

https://gist.github.com/spite/7ae92212b4f28076ba29

And your commit https://github.com/mrdoob/three.js/commit/b428a19e5064bf2ba7506706cd41495ffaadcdf9